test/integration/diag_nodes_test.go

```diff
@@ -55,26 +60,32 @@ func TestDiagNodeConditions(t *testing.T) {
 	}
 
 	// Make the node unschedulable and verify diagnostics notices
-	nodeList.Items[0].Spec.Unschedulable = true
-	if _, err := client.Nodes().Update(&(nodeList.Items[0])); err != nil {
-		t.Fatalf("expected no errors making node unschedulable, but: %#v", err)
-	}
-	result = nodeDiag.Check()
-	if errors := result.Errors(); len(errors) != 1 ||
-		!diagtype.MatchesDiagError(errors[0], "DClu0004") {
-		t.Fatalf("expected 1 error about not having nodes, but: %#v", errors)
-	} else if warnings := result.Warnings(); len(warnings) < 1 || !diagtype.MatchesDiagError(warnings[0], "DClu0003") {
-		t.Fatalf("expected a warning about test-node not being schedulable, but: %#v", warnings)
-	}
+	waitForNode(t, client, 3, "making node unschedulable", func(nodeList *kapi.NodeList, t *testing.T) (bool, error) {
+		nodeList.Items[0].Spec.Unschedulable = true
+		if _, err := client.Nodes().Update(&(nodeList.Items[0])); err != nil {
+			return false, err
+		}
+		result = nodeDiag.Check()
+		if errors := result.Errors(); len(errors) != 1 ||
+			!diagtype.MatchesDiagError(errors[0], "DClu0004") {
+			t.Fatalf("expected 1 error about not having nodes, but: %#v", errors)
+		} else if warnings := result.Warnings(); len(warnings) < 1 || !diagtype.MatchesDiagError(warnings[0], "DClu0003") {
+			t.Fatalf("expected a warning about test-node not being schedulable, but: %#v", warnings)
+		}
+		return true, nil
+	})
 
 	// delete it and check with no nodes defined; should get an error about that.
-	if err := client.Nodes().Delete(nodeList.Items[0].ObjectMeta.Name); err != nil {
-		t.Fatalf("expected no errors deleting node, but: %#v", err)
-	}
-	if errors := nodeDiag.Check().Errors(); len(errors) != 1 ||
-		!diagtype.MatchesDiagError(errors[0], "DClu0004") {
-		t.Errorf("expected 1 error about not having nodes, not: %#v", errors)
-	}
+	waitForNode(t, client, 3, "deleting node", func(nodeList *kapi.NodeList, t *testing.T) (bool, error) {
```

Comment:
true, however waitForNode is the place to get a nodeList at the moment and I don't really want to make its signature uglier just for this.